scala/core/src/partprops/Parts.scala

```diff
@@ -0,0 +1,141 @@
+package semagrams
+
+import upickle.default._
+  
+case class Part(id:UUID,ob:Ob) extends Entity:
+
+  val ty = ob
+
+  override def toString = if ob.label == ""
+    then "Part" + id.rand
+    else ob.label + id.rand
+
+
+  /** Transform to an name that is usable in tikz */
+  def tikzName: String = ob.label + id.toString
+
+
+object Part:
+  def apply(x:Ob) = new Part(UUID("Part"),x)
+  def apply(id:UUID,x:Ob) = new Part(id,x)
+  
+  
+  def rw: ReadWriter[Part] = 
+    readwriter[(UUID,UUID)].bimap[Part](
+      part => (part.ob.id,part.id),
+      (obId,partId) => Part(partId,Table(obId))
+    )
+
+
+
+
+
+/** A trait for the representation of individual parts in an ACSet, defining
+ *  interoperability with `PropMap`s 
+ */
+trait PartData[Data]:
```

Comment:
Where do you use something that's not `PropMap`?

Reply:
I'm not. This could probably just go, but the idea was that it could be an ACSet could also be a PartData

Reply:
Non-trivial refactor to remove.

Leaving for now.